be/src/exec/vectorized/sorted_streaming_aggregator.cpp

```diff
@@ -0,0 +1,419 @@
+        size_t cnt = 0;
+        //
+        for (size_t i = 0; i < _cmp_vector.size(); ++i) {
+            if (_cmp_vector[i] == 0) {
+                _tmp_agg_states[i] = last_state;
+            } else {
+                _tmp_agg_states[i] = batch_allocated_states + cnt++ * _agg_states_total_size;
+            }
+            last_state = _tmp_agg_states[i];
+        }
+
+        // only create the state when selector == 0
+        std::vector<uint8_t> create_selector(chunk_size);
+        for (size_t i = 0; i < _cmp_vector.size(); ++i) {
+            create_selector[i] = _cmp_vector[i] == 0;
+        }
+
+        //
+        for (size_t i = 0; i < _agg_fn_ctxs.size(); i++) {
+            _agg_functions[i]->batch_create_with_selection(_agg_fn_ctxs[i], chunk_size, _tmp_agg_states,
+                                                           _agg_states_offsets[i], create_selector);
+        }
+    }
+
+    bool use_intermediate = _use_intermediate_as_input();
+    // prepare output column
+    // batch_update/merge stage
+    {
+        SCOPED_TIMER(_agg_compute_timer);
+        for (size_t i = 0; i < _agg_fn_ctxs.size(); i++) {
+            if (!_is_merge_funcs[i] && !use_intermediate) {
+                _agg_functions[i]->update_batch(_agg_fn_ctxs[i], chunk_size, _agg_states_offsets[i],
+                                                _agg_input_raw_columns[i].data(), _tmp_agg_states.data());
+            } else {
+                DCHECK_GE(_agg_input_columns[i].size(), 1);
+                _agg_functions[i]->merge_batch(_agg_fn_ctxs[i], _agg_input_columns[i][0]->size(),
+                                               _agg_states_offsets[i], _agg_input_columns[i][0].get(),
+                                               _tmp_agg_states.data());
+            }
+        }
+    }
+
+    // selector[i] == 0 means selected
+    std::vector<uint8_t> selector(chunk_size);
+    size_t selected_size = 0;
+    {
+        SCOPED_TIMER(_agg_compute_timer);
+        for (size_t i = 1; i < _cmp_vector.size(); ++i) {
+            selector[i - 1] = !(_cmp_vector[i] != 0);
+            selected_size += !selector[i - 1];
+        }
+        // we will never select the last rows
+        selector[chunk_size - 1] = 1;
+    }
+
+    // finalize state
+    // group[i] != group[i - 1] means we have add a new state for group[i], then we need call finalize for group[i - 1]
+    auto agg_result_columns = _create_agg_result_columns();
+    {
+        SCOPED_TIMER(_get_results_timer);
+        if (_cmp_vector[0] != 0 && _last_state) {
+            _finalize_to_chunk(_last_state, agg_result_columns);
+        }
+
+        for (size_t i = 0; i < _agg_fn_ctxs.size(); i++) {
+            _agg_functions[i]->batch_finalize_with_selection(_agg_fn_ctxs[i], chunk_size, _tmp_agg_states,
+                                                             _agg_states_offsets[i], agg_result_columns[i].get(),
+                                                             selector);
```

Comment:
I think here we can use _cmp_vector directly, we do not need create a new selector.

Reply:
our selection use selection[i] == 0 as selected. I keep our interfaces consistent.
ref:
```
    void update_batch_selectively(FunctionContext* ctx, size_t chunk_size, size_t state_offset, const Column** columns,
                                  AggDataPtr* states, const std::vector<uint8_t>& filter) const override {
        for (size_t i = 0; i < chunk_size; i++) {
            // TODO: optimize with simd ?
            if (filter[i] == 0) {
                static_cast<const Derived*>(this)->update(ctx, columns, states[i] + state_offset, i);
            }
        }
    }
```